app/vmalert/datasource/client_prom.go

```diff
@@ -222,6 +224,34 @@ func parsePrometheusResponse(resp *http.Response) (res Result, err error) {
 	return res, nil
 }
 
+func parsePrometheusRangeResponse(resp *http.Response) (res Result, err error) {
+	r, err := parsePromResponse(resp)
+	if err != nil {
+		return res, fmt.Errorf("failed to parse response: %w", err)
+	}
+	if r.Data.ResultType != rtMatrix {
+		return res, fmt.Errorf("unexpected result type %q; expected result type %q", r.Data.ResultType, rtMatrix)
```

Comment:
Since it won't return any valid results for non-matrix responses in replay mode, I think we can just exit vmalert immediately instead of making more retries and producing repetitive errors, wdyt?

Reply:
To make the error unrecoverable?